### ISSUE TITLE
[dev.icinga.com #10943] Endpoints created via API cause zone membership error

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#3823]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/10943

**Created by Destreyf on 2016-01-07 20:41:17 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _Backlog_
Last Update: _2016-11-09 14:54:29 +00:00 (in Redmine)_

    Icinga Version: r2.4.1-1
    Backport?: Not yet backported
    Include in Changelog: 1

---

When running the command to create an endpoint, using the following data:

curl -k -s -u root:password -H 'Accept: application/json' -X PUT 'https://localhost:5665/v1/objects/endpoints/christemppc.example.info'

The response is this

{
[results]() \[
{
[code]() 500,
[errors]() \[
"Error: Endpoint 'christemppc.example.info' does not belong to a zone.\\nLocation: in /var/lib/icinga2/api/packages/\_api/masterspy-1448477170-1/conf.d/endpoints/christemppc.example.info.conf: 1:0-1:41\\n"
\],
[status]() "Object could not be created."
}
\]
}

---

**Relations**:

* relates [#10943](https://dev.icinga.com/issues/10943)

**Comment by remy1991** [comment]
Anyone figured out a straightforward way to deal with adding hosts, endpoints and zones? I am trying to use API to add host dynamically and some of my checks are dependant on certain "vars" but command_endpoint isnt working as the endpoint/zone doesnt exist. Cant add endpoint as it fails saying, "Error: Endpoint 'host_name' does not belong to a zone".

**Comment by remy1991** [comment]
Okay. I am resorting to check_by_ssh for hosts added via api then. Anyone has any better method?

**Comment by widhalmt** [comment]
@remy1991 : Yes, create a config package where you put in zone and endpoint at once. This should work (although I haven't tested it yet)

**Comment by relgames** [comment]
Hi @Destreyf did you find a workaround on dynamically creating endpoints? You comment https://github.com/Icinga/icinga2/issues/3823#issuecomment-273095912 seems to suggest you found it, but can you share more details?

How do you configure Zones on Parent Master /Site Master?
How do you configure endpoints?
How do you configure checks? 
What zone and endpoint do you specify for checks?

We've been fighting with the same issue. It's sad to see Icinga devs don't know how to fix it.

**Comment by Al2Klimov** [comment]
@N-o-X Shall we just let the users use config packages? Whereas checkables are "just some leaves" suitable for hotplugging, zones and endpoints are the foundation of our cluster tree and shall not be hotpluggable IMAO.

**Comment by relgames** [comment]
We are trying to implement dynamic addition/removing of Agents and Satellites. 
With API it's just so much simpler. 

Not sure why you want to remove that feature when there is an existing PR which resolved the problem.

**Comment by Al2Klimov** [comment]
_PING_ @N-o-X

**Comment by Al2Klimov** [comment]
It seems I have to re-address [my question](https://github.com/Icinga/icinga2/issues/3823#issuecomment-598336287)...

**Comment by relgames** [comment]
Absolutely, there is a workaround - for example, we moved to Zabbix, it resolved this issue completely.
